crates/topos-tce-storage/src/validator/mod.rs

```diff
@@ -286,10 +286,18 @@ impl ValidatorStore {
         Ok(ids)
     }
 
-    pub fn insert_pending_certificate(
+    pub async fn insert_pending_certificate(
         &self,
         certificate: &Certificate,
     ) -> Result<Option<PendingCertificateId>, StorageError> {
+        // A lock guard is asked during the insertion of a pending certificate
+        // to avoid race condition when a certificate is being inserted and added
+        // to the pending pool at the same time
```

Comment:
```suggestion
        // to avoid race condition when a certificate (C1) is delivered and is in the process of being persisted to disk (so no longer pending), and a dependent certificate (C2) is added
        // to the pending pool even though it could be delivered as well.
```

Reply:
That's not quit true but I will rephrase